### PR TITLE
[IOTDB-3473][metrics] Prometheus reporter will lose data in some conditions

### PR DESCRIPTION
**Pull request by xinzhongtianxia**
see https://issues.apache.org/jira/browse/IOTDB-3473

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/service/metrics/recorder/CompactionMetricsRecorder.java [hunk 1]
```diff
@@ -144,9 +144,7 @@ public static void recordTaskInfo(
                 Metric.COST_TASK.toString(),
                 MetricLevel.IMPORTANT,
                 Tag.NAME.toString(),
-                "compaction",
-                Tag.TYPE.toString(),
-                isInnerTask ? "inner" : "cross");
+                isInnerTask ? "inner_compaction" : "cross_compaction");
```
Comment (SpriCoder) [inline server/src/main/java/org/apache/iotdb/db/service/metrics/recorder/CompactionMetricsRecorder.java:147]: Please update doc at the same time.

**Review by SpriCoder — COMMENTED** [review]
Good Job! Please update doc related at the same time.

File: server/src/main/java/org/apache/iotdb/db/service/metrics/recorder/CompactionMetricsRecorder.java (hunk 1 above)
Comment (xinzhongtianxia) [reply server/src/main/java/org/apache/iotdb/db/service/metrics/recorder/CompactionMetricsRecorder.java:147]: fixed

**Review by SpriCoder — APPROVED** [review]
LGTM~

**Pull request merged**